model/ArrayList.php

```diff
@@ -477,6 +477,72 @@ public function canFilterBy($by) {
 	 *          // aziz with the age 21 or 43 and bob with the Age 21 or 43
 	 */
 	public function filter() {
+
+		$keepUs = call_user_func_array(array($this, 'normaliseFilterArgs'), func_get_args());
+
+		$itemsToKeep = array();
+		foreach($this->items as $item){
+			$keepItem = true;
+			foreach ($keepUs as $column => $value) {
+				if ((is_array($value) && !in_array($this->extractValue($item, $column), $value))
+					|| (!is_array($value) && $this->extractValue($item, $column) != $value)
+				) {
+					$keepItem = false;
+					break;
+				}
+			}
+			if($keepItem) {
+				$itemsToKeep[] = $item;
+			}
+		}
+
+		$list = clone $this;
+		$list->items = $itemsToKeep;
+		return $list;
+	}
+
+	/**
+	 * Return a copy of this list which contains items matching any of these charactaristics.
+	 *
+	 * @example // only bob in the list
+	 *          $list = $list->filterAny('Name', 'bob');
+	 * @example // azis or bob in the list
+	 *          $list = $list->filterAny('Name', array('aziz', 'bob');
+	 * @example // bob or anyone aged 21 in the list
+	 *          $list = $list->filterAny(array('Name'=>'bob, 'Age'=>21));
+	 * @example // bob or anyone aged 21 or 43 in the list
+	 *          $list = $list->filterAny(array('Name'=>'bob, 'Age'=>array(21, 43)));
+	 * @example // all bobs, phils or anyone aged 21 or 43 in the list
+	 *          $list = $list->filterAny(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+	 *
+	 * @param string|array See {@link filter()}
+	 * @return DataList
+	 */
+	public function filterAny() {
+		$keepUs = call_user_func_array(array($this, 'normaliseFilterArgs'), func_get_args());
+
+		$itemsToKeep = array();
+
+		foreach ($this->items as $column => $value) {
+			$keepItem = false;
+			if ((is_array($value) && in_array($this->extractValue($item, $column), $value))
```

Comment:
The matching changes slightly if you want to match any or all, though.

It would be better to fix that, but at the moment I just want to add these APIs in rather than refactor large amounts of `ArrayList`